stitches/dev_matchup.py

```diff
@@ -155,9 +161,28 @@ def calculate_rolling_mean(input_data, size):
   # Index the data frame by the year, so that the rolling mean respects the years
   data.index = data['year']
 
-  # Now calculate the rolling mean that uses a centered window.
-  group_by = ['model', 'experiment', 'ensemble', 'variable']
-  rslt = data.groupby(group_by)['value'].rolling(size, center=True).mean().reset_index()
+  # previously, we just had a call:
+  # rslt = data.groupby(group_by)['value'].rolling(size, center=True).mean().reset_index()
+  # This returns the proper rolling mean on grouped data, but the first and last (size-1)/2
+  # values in each grouped time series return as NaN because the full window isn't available
+  # on both sides of the values there. We want to still have 'semi-smoothed' values there, as
+  # in, we average whatever data we have in each grouped time series. So for a window size=5,
+  # the smoothed 2099 value would be the average of 2097-2100 in our goal set up (as opposed
+  # to 2097-2101 if we had the full 5 years of data centered on 2099).
+  # In theory, pd.rolling(min_periods=1) does exactly what we want; however, it's weird on
+  # grouped data. Sticking with the example of window size = 5, it was bringing in values from the
+  # next group's time series to fill in 2099 and 2100.
+  # To get pandas.rolling(min_periods-1) to apply correctly to grouped data, we have to wrap it
+  # in a call to `transform` and save that as a new column:
+
+  rslt = data.copy()
+  rslt['rollingAvg'] = rslt.groupby(group_by)['value'].transform(
+    lambda x: x.rolling(size, center=True, min_periods=1).mean())
```

Comment:
What is `group_by` defined? It isn't defined anywhere so when I try to use this function I get the following error message 

```Traceback (most recent call last):
  File "/Users/dorh012/anaconda3/envs/pangeo/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3418, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-6cc812964227>", line 1, in <module>
    out1 = matchup.calculate_rolling_mean(t_anomaly, 5)
  File "/Users/dorh012/Documents/2021/stitches/stitches/dev_matchup.py", line 179, in calculate_rolling_mean
    rslt['rollingAvg'] = rslt.groupby(group_by)['value'].transform(
NameError: name 'group_by' is not defined
```

Reply:
I think adding the following  before line 179 

` group_by = ['model', 'experiment', 'ensemble', 'variable']` should do the trick